### PR TITLE
Improve LinkControl preview

### LINKED ISSUE #53654: Explore simpler link previews
Exploring how to make link control previews a bit more predictable/less cluttered. The big visual of the social image is nice (especially when it fits) but it detracts from the primary actions of edit, unlink (and perhaps copy link https://github.com/WordPress/gutenberg/issues/53655) quick controls. 

Key ideas explored here: 
- Leverage truncation in fetched site title.
- Remove fetched social image. 
- Remove fetched site description. 
- Only show site icon/favicon if there is one (no fallback globe icon). 

#### Current: 
current-trunk

#### Proposed: 
Without site icon/favicon: 
Iteration 3

With site icon/favicon: 
Iteration 2, with site icon

---

The following visuals are based on if https://github.com/WordPress/gutenberg/pull/53566 makes it in:

#### Current: 

Without site icon/favicon: 
Iteration 1

With site icon/favicon: 
Iteration 2, with site icon

### PR DESCRIPTION
**Pull request by richtabor**
## What?
Closes https://github.com/WordPress/gutenberg/issues/53654 by reducing the weight of the link previews, encouraging focus on the primary actions of editing and unlinking. Also, with the addition of https://github.com/WordPress/gutenberg/pull/57726, the link popover will stay open upon creating a link—that popover shouldn't cover so much content. 

Note that "Open in a new tab" is still in this preview. However with https://github.com/WordPress/gutenberg/pull/57726 and https://github.com/WordPress/gutenberg/issues/50892 on the horizon, when creating a link, you'll have access to all link editing controls right off, so we can remove it after. 

## Testing Instructions

1. Open a post or page. 
2. Insert a paragraph block with content.
3. Add a link. 
4. Select the link. 

## Screenshots or screencast 

| Before  | After |
| ------------- | ------------- |
|CleanShot 2024-01-11 at 13 57 17|CleanShot 2024-01-18 at 10 05 47|

### Follow ups:
We would need to follow up once https://github.com/WordPress/gutenberg/pull/57726 and https://github.com/WordPress/gutenberg/issues/50892 are completed, to remove the new tab control from the preview: 

CleanShot 2024-01-11 at 13 56 45

### CODE REVIEW

**Comment by richtabor** [comment]
@getdave I'd appreciate help with tests here when you get a few. 🙇

**Comment by jasmussen** [comment]
Looks good!

**Review by getdave — COMMENTED** [review]
I'm concerned about removing the `Opens in new tab` prematurely.

It was added as a stopgap whilst we worked on other solutions to allow immediate access to the link settings following initial creation.

We have _other_ PRs which will remove the `Opens in new tab` setting whilst also providing a new UX to achieve the same result. Removing it in this PR without providing a alternative is going to regress the experience for some users.

My opinion is that we should not do it in _this_ PR.

**Comment by getdave** [comment]
We could also consider retaining the rich image preview _if_ the mime type is an image.

[This PR](https://github.com/WordPress/gutenberg/pull/57824) shows how we could access the mime-type data to determine if the linked resource is an image.

**Comment by richtabor** [comment]
> I'm concerned about removing the Opens in new tab prematurely.

How confident are you in getting https://github.com/WordPress/gutenberg/pull/57726 in? Would having that resolve this concern?

**Comment by getdave** [comment]
@richtabor Removal of the `Open in new tab` checkbox from this screen is based on the fact that 

> > I'm concerned about removing the Opens in new tab prematurely.
> 
> How confident are you in getting #57726 in? Would having that resolve this concern?

No because that PR still shows a preview after creation. It's the next step after that that will unblock this.

**Comment by richtabor** [comment]
> @richtabor Removal of the Open in new tab checkbox from this screen is based on the fact that

I added it back. We can follow up once those other dependencies are merged.

File: packages/block-editor/src/components/link-control/style.scss
Comment (jeryj) [inline packages/block-editor/src/components/link-control/style.scss:194]: This does meet AA level for normal text, but fails at AAA. It is pretty hard to read at this color at such a small text size. Could we increase the contrast here?

File: packages/format-library/src/link/inline.js
Comment (jeryj) [inline packages/format-library/src/link/inline.js]: I prefer the previous offset that gave it a little bit more separation from the link, but not a sticking point for me.

File: packages/block-editor/src/components/link-control/link-preview.js
Comment (jeryj) [inline packages/block-editor/src/components/link-control/link-preview.js:41]: Do we still need `hasRichData`? It adds an `.is-rich` class, but that class is unused in the CSS.

Comment (jeryj) [reply packages/block-editor/src/components/link-control/link-preview.js:41]: Oh, looks like it was only ever used for tests. Not going to rip that out right now.

File: packages/block-editor/src/components/link-control/style.scss
Comment (richtabor) [reply packages/block-editor/src/components/link-control/style.scss:194]: AA is the requirement. I did base this off size/color from existing UI, to not introduce additional design patterns: 

CleanShot 2024-01-23 at 11 58 55

CleanShot 2024-01-23 at 12 00 20

File: packages/format-library/src/link/inline.js
Comment (richtabor) [reply packages/format-library/src/link/inline.js]: Let's meet in the middle.

CleanShot 2024-01-23 at 12 06 11

File: packages/block-editor/src/components/link-control/test/index.js
Comment (MaggieCabrera) [inline packages/block-editor/src/components/link-control/test/index.js:2248]: The test was failing because now we have an extra span wrapping the title in the preview because we are using `<Truncate>`. I don't know how to do this more cleanly! In all honesty, this same test file does this in another instance in the same way.

**Review by jeryj — APPROVED** [review]
🚢

**Comment by t-hamano** [comment]
This PR also seems to have affected the `MediaReplaceFlow` component. Details are summarized in #58593. I would appreciate it if you could confirm that the layout below is what you intended.

| Before this PR is merged | After this PR is merged |
|--------|--------|
| commit_27fc84e0 | commit_614e0355 |

**Pull request merged**
